opslevel/resource_opslevel_scorecard.go

```diff
@@ -10,10 +10,11 @@ func handleInput(d *schema.ResourceData) opslevel.ScorecardInput {
 	description := d.Get("description").(string)
 
 	input := opslevel.ScorecardInput{
-		Name:        d.Get("name").(string),
-		OwnerId:     *opslevel.NewID(d.Get("owner_id").(string)),
-		Description: &description,
-		FilterId:    opslevel.NewID(d.Get("filter_id").(string)),
+		Name:                        d.Get("name").(string),
+		OwnerId:                     *opslevel.NewID(d.Get("owner_id").(string)),
+		Description:                 &description,
+		FilterId:                    opslevel.NewID(d.Get("filter_id").(string)),
+		AffectsOverallServiceLevels: opslevel.Bool(d.Get("affects_overall_service_levels").(bool)),
```

Comment:
Can you test 1 thing for me.  Since this field is optional but right here its mandatory - i'm 99% sure it will work but i think it will have an inadvertent issue with update.  So here is the flow i'd like you to test

- create a scorecard with terraform where `affects_overall_service_levels` is not part of the terraform configuration
- find the scorecard in the UI and manually edit it so that `affects_overall_service_levels` == true
- plan / apply terraform again without changing the configuration and i'm 99% it will reset `affects_overall_service_levels` back to false

Now - if this does do this - its unclear to me what is the expected behaviour.  In my history with terraform if you have an optional field in the schema and you don't set it in your terraform config - terraform won't manage it BUT it also shouldn't ever change that state if someone tweaked it from the UI.  I'm open to just shipping this but I want to make sure we understand what we are doing here and that if this gets raised as a bug in the future we are saying this is working as expected.  That or we make the field required in the terraform schema even though its optional in our API.

Reply:
You called it! Terraform creates the `scorecard` without `affects_overall_service_levels` as expected. After updating the scorecard in the UI, terraform plan shows it wants to update `affects_overall_service_levels` from `true` --> `null` (which will resolve to false).

I think making this field required here would cause fewer headaches and surprises.